bond_calculator.py

```diff
@@ -127,6 +131,8 @@ def calc_modified_duration(self, bond, yld):
         D = self.calc_macaulay_duration(bond, yld)
 
         # TODO: implement details here - Ching Kung
+        result = D / (1 + yld)

```

Comment:
Same as the above, yld need to be divided by the payment freq, just call function `calc_one_period_discount_factor` to get the discount factor and return ` - D * discount_factor`